frontend/src/components/MapView/MapTooltip/index.tsx

```diff
@@ -7,40 +7,117 @@ import {
   WithStyles,
   LinearProgress,
 } from '@material-ui/core';
+import { GeoJSON } from 'geojson';
+import { isEqual } from 'lodash';
 import { tooltipSelector } from '../../../context/tooltipStateSlice';
 import { isEnglishLanguageSelected, useSafeTranslation } from '../../../i18n';
+import { getRoundedData } from '../../../utils/data-utils';
 
-function MapTooltip({ classes }: TooltipProps) {
+const MapTooltip = memo(({ classes }: TooltipProps) => {
   const popup = useSelector(tooltipSelector);
   const { t, i18n } = useSafeTranslation();
-  return popup.showing && popup.coordinates ? (
-    <Popup
-      anchor="bottom"
-      coordinates={popup.coordinates}
-      className={classes.popup}
-    >
-      <h4>
-        {isEnglishLanguageSelected(i18n)
-          ? popup.locationName
-          : popup.locationLocalName}
-      </h4>
-      {Object.entries(popup.data)
-        .filter(([, value]) => value.coordinates === popup.coordinates)
-        .map(([key, value]) => (
-          <>
-            <h4 key={key}>
-              {t(key)}: {value.data}
-            </h4>
+
+  const popupShowing = useMemo(() => {
+    return popup.showing;
+  }, [popup.showing]);
+
+  const popupCoordinates = useMemo(() => {
+    return popup.coordinates;
+  }, [popup.coordinates]);
+
+  const popupLocationName = useMemo(() => {
+    return popup.locationName;
+  }, [popup.locationName]);
+
+  const popupLocationLocalName = useMemo(() => {
+    return popup.locationLocalName;
+  }, [popup.locationLocalName]);
+
+  const popupTitle = useMemo(() => {
+    if (isEnglishLanguageSelected(i18n)) {
+      return popupLocationName;
+    }
+    return popupLocationLocalName;
+  }, [i18n, popupLocationLocalName, popupLocationName]);
+
+  const popupData = useMemo(() => {
+    return popup.data;
+  }, [popup.data]);
+
+  const popupWmsGetFeatureInfoLoading = useMemo(() => {
+    return popup.wmsGetFeatureInfoLoading;
+  }, [popup.wmsGetFeatureInfoLoading]);
```

Comment:
I added small useMemo pure functions to have better handling on the data inside the component and to know any time what exactly is rerendered and not everything. I can remove them of course, if we find it inefficient, I mean yes this makes the file larger and the lines of code but I thought it would be a lot cleaner and readable